### PR TITLE
Remove unhittable methods in internals

### PR DESCRIPTION
**Pull request by jbrockmendel**
And one more 24024 in test cleanup

### CODE REVIEW

File: pandas/core/internals/blocks.py [hunk 1]
```diff
@@ -1507,15 +1507,8 @@ def _nanpercentile(values, q, axis, **kw):
                                        len(values)).reshape(len(values),
                                                             len(qs))
             else:
-
-                try:
-                    result = _nanpercentile(values, np.array(qs) * 100,
-                                            axis=axis, **kw)
-                except ValueError:
-
-                    # older numpies don't handle an array for q
```
Comment (jbrockmendel) [inline pandas/core/internals/blocks.py]: I'm actually just assuming here that the relevant old numpies can by dropped.  @h-vetinari any idea on this?

File: pandas/core/internals/blocks.py [hunk 2]
```diff
@@ -2400,8 +2346,6 @@ def _maybe_coerce_values(self, values, dtype=None):
             values = self._holder(values)
 
         if dtype is not None:
-            if isinstance(dtype, compat.string_types):
-                dtype = DatetimeTZDtype.construct_from_string(dtype)
```
Comment (jbrockmendel) [inline pandas/core/internals/blocks.py]: let the DatetimeArray constructor handle this

File: pandas/core/internals/blocks.py [hunk 3]
```diff
@@ -1700,49 +1686,9 @@ def putmask(self, mask, new, align=True, inplace=False, axis=0,
...
         return [self.make_block(values=new_values)]
 
-    def _slice(self, slicer):
-        """ return a slice of my values (but densify first) """
-        return self.get_values()[slicer]
-
     def _try_cast_result(self, result, dtype=None):
         return result
 
-    def _unstack(self, unstacker_func, new_columns, n_rows, fill_value):
```
Comment (jbrockmendel) [inline pandas/core/internals/blocks.py]: this and the two methods above are in NonConsolodateableMixin, which is now only mixed in to ExtensionBlock, which directly defines these three methods.

**Comment by jreback** [comment]
lgtm. some lint issues. ping on green.

**Comment by jbrockmendel** [comment]
ping

**Comment by jreback** [comment]
thanks!

**Pull request merged**